From a2d39063424a15209645703c963f8e8a24180cba Mon Sep 17 00:00:00 2001
From: Qwinci <32550582+Qwinci@users.noreply.github.com>
Date: Mon, 25 Nov 2024 18:05:36 +0200
Subject: [PATCH 4/6] Add crescent support to libunwind

---
 libunwind/src/RWMutex.hpp | 20 +++++++++++++++++++-
 1 file changed, 19 insertions(+), 1 deletion(-)

diff --git a/libunwind/src/RWMutex.hpp b/libunwind/src/RWMutex.hpp
index 344d35641f..60c8c0ee50 100644
--- a/libunwind/src/RWMutex.hpp
+++ b/libunwind/src/RWMutex.hpp
@@ -15,7 +15,7 @@
 
 #if defined(_WIN32)
 #include <windows.h>
-#elif !defined(_LIBUNWIND_HAS_NO_THREADS)
+#elif !defined(_LIBUNWIND_HAS_NO_THREADS) && !defined(__crescent__)
 #include <pthread.h>
 #if defined(__ELF__) && defined(_LIBUNWIND_LINK_PTHREAD_LIB)
 #pragma comment(lib, "pthread")
@@ -34,6 +34,24 @@ public:
   bool unlock() { return true; }
 };
 
+#elif defined(__crescent__)
+
+class _LIBUNWIND_HIDDEN RWMutex {
+public:
+  bool lock_shared() {
+    return true;
+  }
+  bool unlock_shared() {
+    return true;
+  }
+  bool lock() {
+    return true;
+  }
+  bool unlock() {
+    return true;
+  }
+};
+
 #elif defined(_WIN32)
 
 class _LIBUNWIND_HIDDEN RWMutex {
-- 
2.45.2

