From 0336cc59824611ddcad73abd3c61b610c6004c83 Mon Sep 17 00:00:00 2001
From: Qwinci <32550582+Qwinci@users.noreply.github.com>
Date: Mon, 25 Nov 2024 18:05:57 +0200
Subject: [PATCH 5/6] Add crescent support to libcxxabi

---
 libcxxabi/src/CMakeLists.txt | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/libcxxabi/src/CMakeLists.txt b/libcxxabi/src/CMakeLists.txt
index 84fe2784be..957d039bee 100644
--- a/libcxxabi/src/CMakeLists.txt
+++ b/libcxxabi/src/CMakeLists.txt
@@ -37,7 +37,8 @@ else()
 endif()
 
 if (LIBCXXABI_ENABLE_THREADS AND (UNIX OR FUCHSIA) AND NOT (APPLE OR CYGWIN)
-    AND NOT (${CMAKE_SYSTEM_NAME} MATCHES "AIX"))
+    AND NOT (${CMAKE_SYSTEM_NAME} MATCHES "AIX")
+    AND NOT CRESCENT)
   list(APPEND LIBCXXABI_SOURCES
     cxa_thread_atexit.cpp
   )
-- 
2.45.2

